zanata-war/src/main/java/org/zanata/action/ProjectHomeAction.java

```diff
@@ -438,4 +486,383 @@ public void resetPageData() {
+                }
+                personLocaleRoles.get(person).put(membership.getLocale(),
+                        membership.getRole());
+            }
+        }
+    }
+
+    /**
+     * Clear data related to memberships so that it will be re-loaded fro the
+     * database next time it is needed for display.
+     *
+     * This should be done whenever permissions are changed.
+     */
+    public void clearCachedMembershipData() {
+        // Roles may have changed, so role lists are cleared so they will be regenerated
+        personRoles = null;
+        personLocaleRoles = null;
+        project = null;
+
+        // Person may have no roles left and no longer belong in the list, so
+        // ensure the list of people is refreshed.
+        peopleFilterComparator.clearAllMembers();
+        peopleFilterComparator.sortPeopleList();
+    }
+
+    public List<HPerson> getAllMembers() {
+        final Set<HPerson> people = Sets.newHashSet(getMemberRoles().keySet());
+        people.addAll(getPersonLocaleRoles().keySet());
+        return Lists.newArrayList(people);
+    }
+
+    public boolean isTranslator(HPerson person) {
+        ListMultimap<HLocale, LocaleRole>
+            map = getPersonLocaleRoles().get(person);
+        return map != null && !map.isEmpty();
+
+    }
+    public boolean isMaintainer(HPerson person) {
+        List<ProjectRole> roles = getPersonRoles().get(person);
+        return (roles == null || roles.isEmpty()) ? false :
+            roles.contains(ProjectRole.Maintainer) ||
+                roles.contains(TranslationMaintainer);
+    }
+
+    /**
+     * Check whether a person has any project or locale membership in this project.
+     */
+    public boolean isMember(HPerson person) {
+        return getAllMembers().contains(person);
+    }
+
+    /**
+     * Get display names for all of a person's project and locale roles
+     */
+    public List<String> allRoleDisplayNames(HPerson person) {
+        List<String> displayNames = Lists.newArrayList();
+        displayNames.addAll(projectRoleDisplayNames(person));
+        displayNames.addAll(languageRoleDisplayNames(person));
+        return displayNames;
+    }
+
+    /**
+     * Get a list of the display name for every project-related role for a
+     * person.
+     */
+    public Collection<String> projectRoleDisplayNames(HPerson person) {
+        final Collection<ProjectRole> rolesForPerson = getMemberRoles().get(person);
+        Collection<ProjectRole> roles;
+        if (rolesForPerson == null) {
+            roles = Lists.newArrayList();
+        } else {
+            roles = Lists.newArrayList(rolesForPerson);
+        }
+
+        // Maintainer role includes TranslationMaintainer privileges, so do not
+        // show the lower-permission role.
+        if (roles.contains(Maintainer)) {
+            roles.remove(TranslationMaintainer);
+        }
+
+        return Lists.newArrayList(Collections2.transform(roles,
```

Comment:
Method return type is Collection and Collection2.transform will return a collection. Do you still need to wrap it in a list?